uaclient/cli.py

```diff
@@ -477,6 +562,127 @@ def get_valid_entitlement_names(names: "List[str]"):
     return entitlements_found, entitlements_not_found
 
 
+def action_config(args, cfg, **kwargs):
+    """Perform the config action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    parser = get_parser()
+    subparser = parser._get_positional_actions()[0].choices["config"]
+    valid_choices = subparser._get_positional_actions()[0].choices.keys()
+    if args.command not in valid_choices:
+        parser._get_positional_actions()[0].choices["config"].print_help()
+        raise exceptions.UserFacingError(
+            "\n<command> must be one of: {}".format(", ".join(valid_choices))
+        )
+
+
+def action_config_show(args, cfg, **kwargs):
+    """Perform the 'config show' action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    col_width = str(max([len(x) for x in config.UA_CONFIGURABLE_KEYS]) + 1)
+    row_tmpl = "{key: <" + col_width + "}: {value}"
+    for key in config.UA_CONFIGURABLE_KEYS:
+        print(row_tmpl.format(key=key, value=getattr(cfg, key)))
+
+
+@assert_root
+def action_config_set(args, cfg, **kwargs):
+    """Perform the 'config set' action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    from uaclient.apt import setup_apt_proxy
+    from uaclient.entitlements.livepatch import (
+        configure_livepatch_proxy,
+        configure_snap_proxy,
+    )
+
+    parser = get_parser()
+    config_parser = parser._get_positional_actions()[0].choices["config"]
+    subparser = config_parser._get_positional_actions()[0].choices["set"]
+    try:
+        set_key, set_value = args.key_value_pair.split("=")
+    except ValueError:
+        subparser.print_help()
+        raise exceptions.UserFacingError(
+            "\nExpected <key>=<value> but found: {}".format(
+                args.key_value_pair
+            )
+        )
+    if set_key not in config.UA_CONFIGURABLE_KEYS:
+        subparser.print_help()
+        raise exceptions.UserFacingError(
+            "\n<key> must be one of: {}".format(
+                ", ".join(config.UA_CONFIGURABLE_KEYS)
+            )
+        )
+    # TODO(SC-174: add prompts on proxy changes and honor --assume-yes)
+    if set_key in ("http_proxy", "https_proxy"):
+        kwargs = {set_key: set_value}
+        configure_snap_proxy(**kwargs)
+        # Only set livepatch proxy if livepatch is enabled
+        ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["livepatch"]
+        entitlement = ent_cls(cfg)
+        livepatch_status, _ = entitlement.application_status()
+        if livepatch_status == ua_status.ApplicationStatus.ENABLED:
+            configure_livepatch_proxy(**kwargs)
+    elif set_key in ("apt_http_proxy", "apt_https_proxy"):
+        # setup_apt_proxy is destructive for unprovided values. Source complete
+        # current config values from uaclient.conf before applying set_value.
+        kwargs = {
+            "http_proxy": cfg.apt_http_proxy,
+            "https_proxy": cfg.apt_https_proxy,
+        }
+        kwargs[set_key] = set_value
```

Comment:
This looks like a bug, `set_key` needs the "apt_" chopped off like we do below in unset.

Reply:
Good catch, and looks like I have a unittest gap. Will fix.